### PR TITLE
More verbose status message for failed checks

### PR DESCRIPTION
**Pull request by Fennerr**
### Context
Made changes to [3183](https://github.com/prowler-cloud/prowler/pull/3183), but do not have permissions to commit to this branch.

### Description

Added more verbose messaging, so that it is easier for the user to identify why the check failed, and perform remediation.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Fennerr** [comment]
I also filtered out the OPTION methods. It is common for pre-flight requests to be sent without authentication material - so the OPTION method usually doesn't have an authorizer. But we can remove the filtering out of the OPTION methods if you would like

File: prowler/providers/aws/services/apigateway/apigateway_restapi_authorizers_enabled/apigateway_restapi_authorizers_enabled.py [hunk 1]
```diff
@@ -14,27 +14,60 @@ def execute(self):
...
                     report.status = "PASS"
-                    report.status_extended = f"API Gateway {rest_api.name} ID {rest_api.id} has an authorizer configured at api or methods level."
+                    report.status_extended = f"API Gateway {rest_api.name} ({rest_api.id}) has all methods authorized."
 
             findings.append(report)
 
         return findings
+
+    @staticmethod
+    def array_to_string(arr):
```
Comment (jfagoagas) [inline prowler/providers/aws/services/apigateway/apigateway_restapi_authorizers_enabled/apigateway_restapi_authorizers_enabled.py]: I think there is no need for this function, it is enough if you show the values comma-separated, like:
```python
>>> ", ".join(['A', 'B', 'C'])
'A, B, C'
```

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/services/apigateway/apigateway_restapi_authorizers_enabled/apigateway_restapi_authorizers_enabled.py (hunk 1 above)
Comment (Fennerr) [reply prowler/providers/aws/services/apigateway/apigateway_restapi_authorizers_enabled/apigateway_restapi_authorizers_enabled.py]: I have removed it

**Comment by n4ch04** [comment]
Please @Fennerr  refer to https://github.com/prowler-cloud/prowler/pull/3183#issuecomment-1853881493

**Pull request closed without merging**